sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/csv/CSVSuite.scala

```diff
@@ -1174,4 +1174,25 @@ class CSVSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
         }
       }
   }
+
+  test("Do not partially lose data when parsing float and double") {
```

Comment:
I suggest a better description for this test and please include the JIRA number. E.g., 
```SPARK-21263: Invalid float and double are handled correctly in different modes```